tsconfig.json

```diff
@@ -31,6 +32,7 @@ Visit https://aka.ms/tsconfig.json for a detailed list of options.
 
     /* Type Checking */
     "strict": true,                                      /* Enable all strict type-checking options. */
+    "strictNullChecks": false,                           /* Allow unhandled false/null/undefined values to make incremental TypeScript adoption easier. */
```

Comment:
Eslint has the option of disabling a rule for a particular file - does tsc / typescript-eslint have the same? If so, that may be better as we can add a "todo" comment in those files and then incrementally move to adhering to this rule

Reply:
I haven't read the whole Typescript issue thread, but the answer to this seems to be somewhere between "no" and "kinda via a PITA workaround". See [this SO question](https://stackoverflow.com/questions/63894511/can-you-turn-down-strict-mode-et-al-on-a-single-file-or-code-block) which leads to [a Typescript issue](https://github.com/Microsoft/TypeScript/issues/28306).

Reply:
Thanks for digging @genehack. The closest thing I could find in TypeScript docs is [`ts-nocheck`](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check) which means "disable all rules for this file" (not great).

Reply:
Another bad alternative is the per-line `ts-ignore`/`ts-expect-error`:

1. This disables all rules for the line, which is similarly not great
2. Doing this for 443 violations of `strictNullChecks` seems overkill

Reply:
I can update the commit message of 8a0f4fa043297e7a8a5b1b22f77279cc613f8c0a to note that we've considered these alternatives.

Reply:
Thanks for the responses - pity about the answers tho! 

> I can update the commit message of https://github.com/nextstrain/auspice/commit/8a0f4fa043297e7a8a5b1b22f77279cc613f8c0a to note that we've considered these alternatives.

Sounds good. Ideally we would turn this rule on in the future as I think it'll catch a whole lot of bugs. 443 is daunting tho!

Reply:
An alternative is to use a baseline (a concept I discovered [elsewhere](https://github.com/nextstrain/augur/issues/1472#issuecomment-2452297580)). There's no official TypeScript support, but a 3rd party tool [tsc-baseline](https://github.com/TimMikeladze/tsc-baseline) seems promising. Added: 8992782bf1ebf8cb1a104567433ac00625224201

A slight downside is that, since tsc-baseline has not gained much popularity, there is no support beyond the command line. This means code editors such as VS Code's TypeScript integration will use `tsc` and still report all the rule violations. I wouldn't consider this blocking though. Here's a workaround: 1960d5a3f9dc55a7946c9129ba67732db09081d9

Reply:
I'll defer to @genehack but the added complexity of `tsc-baseline` seems too much, although I've not used it enough to make a call here.

Reply:
`tsc-baseline` was worth the shot but I don't think it's the right move.

Added a post-merge task:

> - Post-merge: create issue to address `strictNullChecks` violations (started with 260ba0020e78d7ea4a071c290d32fc2a53e67732)